### PR TITLE
untap: add --force switch

### PR DESCRIPTION
**Pull request by dawidd6**
Another take on: https://github.com/Homebrew/brew/pull/9422

Changes in comparison to previous PR:
- if HOMEBREW_DEVELOPER env variable is set and there are tap formulae or casks installed, print warning instead of dying
- assing formulae and casks name to variable, so we avoid duplicated logic

- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/HEAD/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/brew/blob/HEAD/Library/Homebrew/test/PATH_spec.rb).
- [x] Have you successfully run `brew style` with your changes locally?
- [ ] Have you successfully run `brew tests` with your changes locally?
- [x] Have you successfully run `brew man` locally and committed any changes?

-----

### CODE REVIEW

**Comment by reitermarkus** [comment]
> if HOMEBREW_DEVELOPER env variable is set and there are tap formulae or casks installed, print warning instead of dying

I find this logic is inverse to anything else switched by `HOMEBREW_DEVELOPER`. If I am a `HOMEBREW_DEVELOPER`, I want a hard failure instead of a warning so I, as a developer, can submit a PR to fix the error.

**Comment by reitermarkus** [comment]
We should make this:

1. A warning for everyone.
2. A failure for `HOMEBREW_DEVELOPER`s, a warning for everyone else.
3. A failure for everyone.

File: Library/Homebrew/cmd/untap.rb [hunk 1]
```diff
@@ -28,6 +28,24 @@ def untap
       tap = Tap.fetch(tapname)
       odie "Untapping #{tap} is not allowed" if tap.core_tap?
 
+      installed_tap_formulae = Formula.installed.select { |formula| formula.tap == tap }
+      installed_tap_casks = Cask::Caskroom.casks.select { |cask| cask.tap == tap }
+
+      if installed_tap_formulae.length.positive? || installed_tap_casks.length.positive?
```
Comment (reitermarkus) [inline Library/Homebrew/cmd/untap.rb:34]:
```suggestion
      if installed_tap_formulae.any? || installed_tap_casks.any?
```

Comment (MikeMcQuaid) [reply Library/Homebrew/cmd/untap.rb:34]:
```suggestion
      if installed_tap_formulae.present? || installed_tap_casks.present?
```

**Review by MikeMcQuaid — APPROVED** [review]

**Review by MikeMcQuaid — APPROVED** [review]

**Review by reitermarkus — DISMISSED** [review]
https://github.com/Homebrew/brew/pull/9489#issuecomment-742433652

**Comment by MikeMcQuaid** [comment]
> * A warning for everyone.
> * A failure for `HOMEBREW_DEVELOPER`s, a warning for everyone else.
> * A failure for everyone.

I disagree. The way this is implemented in the PR is consistent with how `brew uninstall` behaves wrt. dependencies.

**Pull request merged**